templates/referral_campaign.html

```diff
@@ -0,0 +1,453 @@
+  }
+  #testing {}
+
+  .hidden-desktop {}
+  .hidden-mobile {
+    display: none;
+  }
+  .desktop-split {}
+  .float-desktop {}
+  .avatar-container {}
+  .avatar {
+    float: left;
+    height: 80px;
+    width: 80px;
+    border: 1px #fff solid;
+    border-radius: 40px;
+    margin: 5px;
+    padding: 0;
+    overflow: hidden;
+    background-position: center;
+    background-size: cover;
+  }
+  .avatar img {
+    /*max-height: 80px;
+    max-width: 80px;*/
+  }
+  .avatar-container p {
+    display: block;
+    margin: 0 auto;
+    padding-top: 15px;
+    font-family: "Lato" sans-serif;
+    font-size: 20px;
+    line-height: 1.3;
+    text-align: left;
+  }
+
+  @media only screen and (min-width: 600px) {
+    .partner-campaign-page {
+      padding-bottom: 50px;
+      clear: both;
+    }
+    #screenshot1 hr,
+    #screenshot2 hr {
+      display: none;
+    }
+    #tagline p { font-size: 25px; }
+    #callout { font-size: 32px }
+    #callout .value { font-size: 88px }
+    #intro p { font-size: 18px; }
+    #partner-campaign-header {
+      margin-top: 100px;
+      margin-left: 30px;
+    }
+    #partner-campaign-header .origin.logo {}
+
+    #screenshot1 img,
+    #screenshot2 img,
+    #screenshot2-desktop img { left: 75px; }
+
+    .avatar {
+      margin: 25px;
+    }
+    .avatar-container p {
+      padding-top: 30px;
+    }
+    .qrcode-wrapper p {
+      font-family: Lato, sans-serif;
+      font-size: 14px;
+      line-height: 26px;
+      text-align: center;
+    }
+    .qrcode-wrapper img {
+      margin: 0 auto;
+      margin-top: 10px;
+    }
+    .qrcode-wrapper {
+      background: #fff;
+      color: #000;
+      padding: 5px;
+      margin: 0 auto;
+      width: 190px;
+      height: 190px;
+      border-radius: 20px;
+    }
+    .hidden-mobile {
+      display: inherit;
+    }
+    .hidden-desktop {
+      display: none;
+    }
+    .desktop-split {
+      clear: both;
+      display: block;
+      max-width: 1024px;
+      margin: 0 auto;
+    }
+    .float-desktop {
+      float: left;
+      width: 50%;
+      margin: 0;
+    }
+    .page-splitter {
+      width: 100%;
+      margin: 50px auto;
+    }
+  }
+</style>
+{% endblock %}
+
+{% block extra_scripts %}
+  <script src="/static/js/qrcode.min.js" type="text/javascript"></script>
+  <script>
+    const isMobile = /iPhone|iPad|iPod|Android/i.test(navigator.userAgent)
+    const isIOS = navigator.platform && /iPad|iPhone|iPod/.test(navigator.platform)
+    const referalCode = `{{ referral_code }}`
+    const me = `${location.origin}/referral/{{ referral_code }}`
+
+    const qrOptions = {
+      text: me,
+      width: 140,
+      height: 140,
+      colorDark : "#000000",
+      colorLight : "#ffffff",
+      correctLevel : QRCode.CorrectLevel.H
+    }
+    const qrcode1 = new QRCode('qrcode1', qrOptions)
+    const qrcode2 = new QRCode('qrcode2', qrOptions)
+
+    /**
+     * Copy a string to the clipboard
+     */
+    function codeToClipboard(code) {
```

Comment:
nit: a lot of this code is already in the [partner_campaign](https://github.com/OriginProtocol/origin-website/blob/48ff064f9dfbca2920690c32ead51027a6918974/templates/partner_campaign.html) Could be separated into a third file and imported from there into this and `partner_campaign` file.

Reply:
Went for speed, not cleanliness.  If these look like they'll stick around, I'll take the time to clean it up a bit.

Reply:
sounds good 👍